blockprotocol/blockprotocol
# Implement fade-in on viewport at `/wordpress`, `/hub` & `/homepage`

## 🌟 What is the purpose of this PR?

This PR implements a fade-in animation on main elements of home page, /wordpress and /hub pages. So elements fade-in once when they are visible to the user (shown at viewport). 

Also to keep the page loading time shorter, implemented lazy loading for `framer-motion`. They explain using a lazy loading version of the `motion` component on their [official docs](https://www.framer.com/motion/guide-reduce-bundle-size/) to reducing the bundle size.


## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204000740778935/1204162407205361/f) _(internal)_
- [Asana task](https://app.asana.com/0/1204000740778935/1204165363163231/f) _(internal)_

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Confirm that /hub, /wordpress & home pages have fade-in-on-scroll animation on elements.

## 📦 Lazy Loading - Bundle Size Demo

Check [this commit](https://github.com/blockprotocol/blockprotocol/pull/1203/commits/8cf004e8bed18563ba0197ce23cc2ba9a8ae8ab9) to see lazy-loading implementation separately

| ![](https://user-images.githubusercontent.com/39553853/226626400-8a6f48c1-5c67-44b5-845a-fa6580c2629f.png)            | ![](https://user-images.githubusercontent.com/39553853/226626845-42af701e-b83f-4c59-8e0a-f6e41d5d8b2e.png)            | ![](https://user-images.githubusercontent.com/39553853/226626880-044cf82e-efd6-4e5d-ba39-f5c19c2161ed.png)            |
|:-------------------------------------:|:-------------------------------------:|:-------------------------------------:|
|Build without `framer-motion`|Build with `framer-motion`                                       | Build with lazy `framer-motion`                                      |
|  app.js = 353 kB | app.js = 383 kB  | app.js = 358 kB  |
|   |   | framer-features = 16.1 kB |

## 📹 Demo


https://user-images.githubusercontent.com/39553853/226611173-4e18ae55-738d-4069-8ef6-dda27bbbea05.mov
